src/main/java/filter/expression/Qualifier.java

```diff
@@ -214,16 +192,16 @@ public static void setCurrentTime(LocalDateTime dateTime) {
     }
 
     public boolean isEmptyQualifier() {
-        return name.isEmpty() && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.EMPTY;
     }
 
     public boolean isFalseQualifier() {
-        return name.equalsIgnoreCase("false") && content.isPresent() && content.get().isEmpty();
+        return type == QualifierType.FALSE;
```

Comment:
Just for consideration: would it be better to compare this and empty qualifier with `Qualifier.FALSE` and `Qualifier.EMPTY` instead? So if there is a bug where someone made uses an `QualifierType.EMPTY` or `QualifierType.FALSE` with content, they won't pass these two.